osquery/worker/system/linux/memory.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include "memory.h"
+
+#include <malloc.h>
+
+#include <osquery/core/flags.h>
+#include <osquery/core/watcher.h>
+#include <osquery/filesystem/linux/proc.h>
+#include <osquery/logger/logger.h>
+
+namespace osquery {
+DECLARE_uint64(malloc_trim_threshold);
+
+#ifdef OSQUERY_LINUX
+
+void releaseRetainedMemory() {
+  auto current_limit = getWorkerLimit(WatchdogLimitType::MEMORY_LIMIT);
+  auto used_memory_res = getProcRSS("self");
+
+  if (used_memory_res.isError()) {
+    VLOG(1) << "Failed to get retained memory: "
+            << used_memory_res.takeError().getMessage();
+    return;
+  }
+
+  auto used_memory = used_memory_res.take() / (1000 * 1000);
+
+  if (current_limit == 0 ||
```

Comment:
So a value of `0` means _always_ try to trim? I would have expected `0` to disable the feature

Reply:
You're right, I think I confused myself with this.